v3/src/components/free-tile-row.tsx

```diff
@@ -14,24 +14,95 @@ interface IFreeTileRowProps {
   getTile: (tileId: string) => ITileModel | undefined
 }
 export const FreeTileRowComponent = observer(({ content, row, getTile }: IFreeTileRowProps) => {
+  const [resizingTileStyle, setResizingTileStyle] =
+    useState<{left: number, top: number, width: number, height: number}>()
+  const [resizingTileId, setResizingTileId] = useState("")
+  const tempWidth = useRef<number>(0)
+  const tempHeight = useRef<number>(0)
+  const tempLeft = useRef<number>(0)
+
   const handleCloseTile = (tileId: string) => {
     if (!tileId) return
     content?.deleteTile(tileId)
   }
+
+  const handleResizePointerDown = (e: React.PointerEvent, tile: IFreeTileLayout, direction: string) => {
+    const startWidth = tile.width
+    const startHeight = tile.height
+    const startPosition = {x: e.pageX, y: e.pageY}
+    tempLeft.current = tile.x
+    tempWidth.current = startWidth
+    tempHeight.current = startHeight
+    let resizingWidth = startWidth, resizingHeight = startHeight, resizingWidthLeft = tile.x
+    // Because user can start drag 8px within the border, the component's startPosition.x moves by number of pixels
+    // the pointer down event location, which moves the entire component to the right by the same number of pixels.
+    // So we force it to always be the left position of the component
+    const leftStartPosition = startPosition.x > tile.x ? tile.x : startPosition.x
+
+    const onPointerMove = (pointerMoveEvent: { pageX: number; pageY: number }) => {
+      setResizingTileId(tile.tileId)
+      switch (direction) {
+        case "bottom-right":
+          resizingWidth = startWidth - startPosition.x + pointerMoveEvent.pageX
+          resizingHeight = startHeight - startPosition.y + pointerMoveEvent.pageY
+          break
+        case "bottom-left":
+          resizingWidth = startWidth + leftStartPosition - pointerMoveEvent.pageX
+          resizingHeight = startHeight - startPosition.y + pointerMoveEvent.pageY
+          resizingWidthLeft = pointerMoveEvent.pageX
+          break
+        case "left":
+          resizingWidth = startWidth + leftStartPosition - pointerMoveEvent.pageX
+
+          resizingWidthLeft = pointerMoveEvent.pageX
+          break
+        case "bottom":
+          resizingHeight = startHeight - startPosition.y + pointerMoveEvent.pageY
+          break
+        case "right":
+          resizingWidth = startWidth - startPosition.x + pointerMoveEvent.pageX
+          break
+      }
+
+      setResizingTileStyle({left: resizingWidthLeft, top: tile.y, width: resizingWidth, height: resizingHeight})
+      tempLeft.current = resizingWidthLeft
+      tempWidth.current = resizingWidth
+      tempHeight.current = resizingHeight
+    }
+    const onPointerUp = () => {
+      document.body.removeEventListener("pointermove", onPointerMove)
+      document.body.removeEventListener("pointerup", onPointerUp)
```

Comment:
I think we should capture the events so as to minimize the chance that any other event handlers can interfere:
```suggestion
      document.body.removeEventListener("pointermove", onPointerMove, { capture: true })
      document.body.removeEventListener("pointerup", onPointerUp, { capture: true })
```